src/App.tsx

```diff
@@ -1,17 +1,129 @@
-import React from 'react';
-import './App.css';
+import React, { useState } from 'react';
+import './App.scss';
+import { Todos } from './types/Todos';
+import { TodoList } from './components/TodoList/TodoList';
 
+import todos from './api/todos';
 import users from './api/users';
 
 const App: React.FC = () => {
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState('');
+  const [todo, setTodo] = useState(todos);
+  const [status, setStatus] = useState('not completed');
+  const [errUser, setErrUser] = useState('');
+  const [errTitle, setErrTitle] = useState('');
+
+  const validate = () => {
+    if (!user) {
+      setErrUser('Please choose a user');
+    }
+
+    if (!title.trim()) {
+      setErrTitle('Please enter the title');
+    }
+
+    if (!user || !title.trim()) {
+      return false;
+    }
+
+    return true;
+  };
+
+  const addTodo = (event: React.SyntheticEvent) => {
+    event.preventDefault();
+    if (validate()) {
+      const searchingUser = users.find(({ name }) => name === user);
+
+      const newTodo = {
+        id: todo[todo.length - 1].id + 1,
+        title,
+        userId: searchingUser ? searchingUser.id : 0,
+        completed: status === 'completed',
+      };
+
+      setTodo((current) => [...current, newTodo]);
+      setTitle('');
+      setUser('');
+      setStatus('not completed');
+    }
+  };
+
+  const preparedTodos: Todos[] = todo.map((oneTodo) => ({
```

Comment:
try to use [useMemo](https://reactjs.org/docs/hooks-reference.html#usememo)
Optional 😉